### PR TITLE
Add a function to delete a time interval from GTI

### PR DESCRIPTION
**Pull request by cgalelli**
As requested by the issue #4784 I added a function that removes a "bad time interval" from the GTIs with correlated test. A future addition will be a more complex handling of lists of "bad periods"

### CODE REVIEW

**Comment by registerrier** [comment]
The CI issue is corrected. You can rebase to make it run.

**Review by registerrier — COMMENTED** [review]
Thanks @cgalelli . This looks good!

I think there is one issue because you don't copy the parent Table. You are probably modifying the GTI.

Otherwise you might add a test when the bad interval is partly outside or fully outside the range of your GTI?

File: gammapy/data/gti.py [hunk 1]
```diff
@@ -318,6 +318,35 @@ def select_time(self, time_interval):
 
         return self.__class__(gti_within)
 
+    def delete_interval(self, time_interval):
+        """Select and crop GTIs in time interval.
+
+        Parameters
+        ----------
+        time_interval : `astropy.time.Time`
```
Comment (bkhelifi) [inline gammapy/data/gti.py:326]: The docstring says that this is an interval, but the type is just one Time. Should not it be a [Time, Time]?

**Review by registerrier — APPROVED** [review]
Thanks @cgalelli ! Looks good to me.

**Pull request merged**